File: Medium/198. House Robber/HouseRobber.java
/*
LeetCode Medium: 198. House Robber
https://leetcode.com/problems/house-robber/description/

Check HouseRobber.cpp for brief solution explanation.
*/
class Solution {
    public int rob(int[] nums) {
        int N = nums.length;
        int[]dp = new int[N+1];
        dp[1] = nums[0];
        for(int i = 2;i<=N;i++){
            dp[i] = Math.max(dp[i-1], nums[i-1]+dp[i-2]);
        }
        return dp[N];
    }
}

File: Medium/198. House Robber/HouseRobber.py
'''
LeetCode Medium: 198. House Robber
https://leetcode.com/problems/house-robber/description/

Check HouseRobber.cpp for brief solution explanation.
'''
class Solution:
    def rob(self, nums: List[int]) -> int:
        N = len(nums)
        dp = [0 for i in range(N+1)]
        dp[1] = nums[0]
        for i in range(2, N+1):
            dp[i] = max(dp[i-1], nums[i-1]+dp[i-2])
        return dp[N]